### ISSUE TITLE
super, ale preco ho prehladavas pri dalsom prvku z prveho pola ? :)

### ISSUE THREAD

**Issue by tu1ly** [vendybike/topMonks#2]
https://github.com/vendybike/topMonks/blob/8e2d11e5504bc5809a78c6d0769db1985de749be/index.js#L14

**Comment by vendybike** [comment]
Kvuli tady tomuto je tam podminka na zacatku, kdyby tam similarity mela byt zase 1, uz takovy vysledek existuje v Mape -> takze takovy pripad nenastane. Pro dalsi (rozdilny) prvek v prvnim poli muze byt ta similarity treba 0.8 a porad muze teoreticky byt nejvetsi ze vsech ostatnich, k danemu stringu.

**Comment by tu1ly** [comment]
ale viac ako 1 byt uz nemoze => ak je viac 1thovych = absolutnych zhod - vsetky su ekvivalentne => tuto konkretnu mozem povazovat za 'vybavenu'

**Comment by tu1ly** [comment]
tj tento element uz nemusim porovnavat v dalsom priechode - tj mam o 1*(pocet zostavajucich prvkov v prvom) menej porovnavani a v idealnom pripade sa viem dostat az na polovicu priechodov

**Comment by vendybike** [comment]
Ano, jak jsem psal v jinych komentarich, jestli bych mel zadane, ze musi byt odbavene vsechny prvky A neni rozdilny pocet prvku v poli, takze vzdy mam priradit jen ke kazdemu prvku jen jeden prvek, tak pak by to souhlasilo.

**Comment by vendybike** [comment]
Haha, tak jsem se nad tim jeste dnes zamyslel, asi to nebude uplne nejtrivialnejsi zpusob to delat takto, mam moznost, nahradit ten prvek hodnotou `null` a tim padem tam dat podminku jestli je hodnota `null` tak ji preskocit a nevysetrovat. 
ale stejne bych iteroval celym polem. Kdyby oddelaval prvky v poli, tak tim, ze to pole neni nijak serazene, tak mi to rozhodi indexy ve vysledne mape. 
Napadlo me jeste reseni, kdy bych vyuzival asociativniPole, kde ty indexy zustanou stejne i po odmazani prvku z toho pole. Rikal jsi, ze dnes odjizdis na dovolenou, mam to tedy zkouset implementovat dal ?

**Comment by tu1ly** [comment]
```mam to tedy zkouset implementovat dal ?``` netreba - islo  o tu debatu okolo toho, nie konkretnu implementaciu
